### ISSUE TITLE
Integration of transport application into the CSIS

### ISSUE THREAD

**Issue by ghilbrae** [clarity-h2020/csis#122]
We are ready to start testing the integration of the transport application into the CSIS. What do you need from us? 

As per today's meeting, it seems that the first step would be to provide a link to the app?

**Comment by rapto** [comment]
The standalone application is ready to test in https://clarity.saver.red 
Authentication is performed by the CAS at https://profile.myclimateservices.eu/cas/
Authorization is performed by the django application at saver.red, so some (now manual) sync is needed between both.

**Comment by luis-meteogrid** [comment]
You may use the following url to access the exposure map: https://clarity.saver.red/mapa_elementos/
I have used my user in CSIS to access that link

**Comment by p-a-s-c-a-l** [comment]
The application should be embedded as iFrame into CSIS. 

I've created the respective [Extended iFrame Entity](https://csis.myclimateservice.eu/node/1982), modified the [Exposure Evaluation template - transport](https://csis.myclimateservice.eu/node/1811) and updated the [A-2 Guadalajara](https://csis.myclimateservice.eu/study/55/step/1954/view/external) study.

**Important:** The iFrame is currently blocked by the SAMEORIGIN policy, you have to remove `X-Frame-Options: "SAMEORIGIN"`.

To provide the username to the application there are two possibilities: query parameters and [semaless.js](https://github.com/travist/seamless.js/).

I would recommend to implement support for the required query parameters in your app, then we can call it with e.g. https://clarity.saver.red/?csis-user=demo. This requires also some work at CSIS side (extracting user name, setting, the parameter). I would implement that in [csis_iframe_connector.js](https://github.com/clarity-h2020/csis-helpers-module/blob/dev/js/csis_iframe_connector.js), once we know the list of required parameters and their content.

The other possibility is seamless.js, it is a bit more complicated to set-up an requires the usage of the seamless.js client library in you app. The main advantage is that it supports *bidirectional* communication between the parent frame (CSIS Drupal) and your app. ATM, I don't think that we would need that.

**Comment by rapto** [comment]
I have included Content-Security-Policy information in order to override the SAMEORIGIN setting (as it seems to be more current). The user issue is pending.

**Comment by p-a-s-c-a-l** [comment]
Still getting

``Refused to display 'https://profile.myclimateservices.eu/cas/login?service=https%3A%2F%2Fclarity.saver.red%2Faccounts%2Flogin%2F%3Fnext%3D%252Fmapa_elementos%252F' in a frame because it set 'X-Frame-Options' to 'sameorigin'.``

at https://csis.myclimateservice.eu/study/55/step/1954/view/external

**Comment by fgeyer16** [comment]
Sorry for the longsilence.
Yes I have to set some csp headers. Will be done next days

**Comment by fgeyer16** [comment]
profile.myclimateservcies.eu now allows to be embedded by csis. Now I can see the login mask of profile in the iframe but I failto login as cas test user (logged in as non cas user in CSIS). My CAS test user is not able to see the esternal tab
How is CAS configured at clarity.saver.red?

**Comment by p-a-s-c-a-l** [comment]
Furthermore, the branding of the transport application (colour, but more importantly the name) should be aligned to the [CLARITY Theme](https://github.com/clarity-h2020/clarity-theme).

**Comment by p-a-s-c-a-l** [comment]
The iFrame on [this page](https://csis.myclimateservice.eu/study/55/step/1952/view/maps) is still not working in **Firefox**:

`Laden verboten durch X-Frame-Options: "SAMEORIGIN" von "https://clarity.saver.red/mapa_elementos/", Website erlaubt keine quellübergreifende (cross-origin) Frames von "https://csis.myclimateservice.eu/study/55/step/1952/view/maps".`

**Comment by fgeyer16** [comment]
On my firefox  the map seems to be shown:
![Bildschirmfoto von »2020-01-23 21-07-04«](https://user-images.githubusercontent.com/25331450/73019801-9b17e280-3e24-11ea-84ff-cc7341a1b10c.png)

Firefox 68.4.1esr on Debian 9

> Content Security Policy: 'x-frame-options' wird wegen 'frame-ancestors'-Direktive ignoriert.
This seems to be a Browser version problem. csp frame-ancestor  is the newer headerto use since the x-frame options seems to be a little cumbersome. (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options)

**Comment by fgeyer16** [comment]
Anyway the X-Frame-Options: "SAMEORIGIN" headershould be removedfrom clarity.saver.red.

**Comment by p-a-s-c-a-l** [comment]
I'm using Firefox 72.0.2 (64-Bit).

**Comment by fgeyer16** [comment]
what says security.csp.enable at about:config in firefox?

**Comment by p-a-s-c-a-l** [comment]
![grafik](https://user-images.githubusercontent.com/1788226/73163757-e6feac00-40f0-11ea-8e27-8a2015bf3918.png)

**Comment by p-a-s-c-a-l** [comment]
Closing in favour of https://github.com/clarity-h2020/csis/issues/134
